torch/_torch_docs.py

```diff
@@ -736,11 +736,16 @@ def parse_kwargs(desc):
 
 add_docstr(torch.cat,
            r"""
-cat(seq, dim=0, out=None) -> Tensor
+cat(seq, dim=0, pad=False, pad_value=0, out=None) -> Tensor
 
 Concatenates the given sequence of :attr:`seq` tensors in the given dimension.
-All tensors must either have the same shape (except in the concatenating
-dimension) or be empty.
+Empty tensors will be ignored.
+If padding is set to false, all tensors must have the same shape
+(except in the concatenating dimension).
+If padding is set to true, all tensors do not need to have the same shape
+(except in the concatenating dimension). Except the concatenating dimension,
+every dimension of every input tensor will be expanded to the largest size of
+this dimension owned by attr:`seq` tensors through padding :attr:`pad_value`.
```

Comment:
tensors don't "own" dimensions.  I'd rearrange this a bit to say "The size of the resulting tensor is the size of the sum of the dimensions in the concatenating dimension, and the maximum of the dimension sizes in all other dimensions.  If an input tensor needs to be logically expanded to fill out its place in the resulting tensor, it is padded with :attr:`pad_value`.

You then also want to talk about the case where a tensor doesn't have that dimension (and give an example below).